sdk/python/kfp/dsl/_container_op.py

```diff
@@ -1046,8 +1078,15 @@ def __init__(
         sidecars=sidecars,
         is_exit_handler=is_exit_handler)
 
-    if not ContainerOp._DISABLE_REUSABLE_COMPONENT_WARNING and '--component_launcher_class_path' not in (
-        arguments or []):
+    # Check if the pipeline is being compiled for v2 spec.
+    self.is_compiling_for_v2 = False
```

Comment:
Instead of doing this, assume compiling_for_v2 is always True and remove this altogether. The only downside is the resource spec for gpus below, which we can validate when submitting to the backend.

Reply:
Done